### PR TITLE
MULE-20054: Optimize decoration of input streams

### PR DESCRIPTION
**Pull request by asanguinetti**
It has been observed that change #9347 introduced a performance regression of about 3~4%.
Upon closer inspection, it seems most of the regression came from the fact that we are now decorating with `ObjectArgumentResolverDecorator` a lot of arguments that we were not decorating before (for example, arguments of type `List`). This change was done so that these arguments were also decorated by the new `CursorComponentDecoratorFactory`.

However, the `ObjectArgumentResolverDecorator` was added on #8528 because some arguments of `Object` type required cursor resolution.

For many of the new arguments that we now want to decorate because of the `CursorComponentDecoratorFactory` (like `List`), we don't really seem to be needing the cursor resolution. I'm proposing to use a different `ArgumentResolverDecorator` that decorates with `CursorComponentDecoratorFactory` but does not perform cursor resolution.

Moreover, it doesn't seem that we need to decorate all arguments with this, only those that are annotated as `Content`, because we were using `NO_OP_INSTANCE` as `CursorComponentDecoratorFactory` for everything else anyway.

[Here](https://stats.perf.anypoint.mulesoft.com/goto/8c61069eaa233dcb026d9c9eded18157) you can see the result of the performance executions of these same changes applied on a 4.4.0-20211129 (that one was selected for stability). The version labeled **MULE-20054b** would be containing only the first commit of this PR (0e175ce) and not the second one (b022efb).

Version | TPS | % to baseline
-- | -- | --
4.4.0-20211129 | 10492 | 0.00%
4.4.0-20211129 MULE-20054b | 10754 | 2.50%
4.4.0-20211129 MULE-20054 | 10816 | 3.09%

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/MethodArgumentResolverDelegate.java [hunk 1]
```diff
@@ -356,8 +362,10 @@ private void initArgumentResolvers() {
           }
         }
       }
-    } else {
+    } else if (Object.class.equals(argumentType)) {
```
Comment (fsgonz) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/MethodArgumentResolverDelegate.java:365]: MULE-17903 to verify why this was done, as indicated in the PR full explanation.

**Review by fsgonz — APPROVED** [review]

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/loader/java/type/ExtensionParameter.java [hunk 2]
```diff
@@ -134,6 +135,14 @@ default boolean isRequired() {
     return java.util.Optional.empty();
   }
 
+  /**
+   * @return Whether this parameter is considered content.
+   */
+  default boolean isContent() {
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/loader/java/type/ExtensionParameter.java:141]: this shouldn't be done here nor it should depend on the extension code but the extension model

**Review by marianogonzalez — CHANGES_REQUESTED** [review]
This solution is trying to optimize #9234 . The problem is that had I seen that PR originally, I woudl have rejected it:

1) The entire solution is entirely based on specific resolvers of the Java SDK. This means that it won't work any of the other SDK flavours 
2) It's based on introspecting the extension's code. It should instead be inspecting the resulting ExtensionModel.
3) it forces adding similar logic for sources, functions, etc. This should all be centralized on the returning piece of it, not on each usage.
4) it forces propagating concepts such as content to components that should not have it

The correct solution here is to revert #9234 and reimplement it completely. This solution does not scale and does not align with the refactors we're doing in the SDK to do reuse. Please write/refer to a spec in which the design for this was discussed.

Thank you

**Comment by asanguinetti** [comment]
I'm closing this PR because we are going to go with the revert option.

**Pull request closed without merging**